### DISCUSSION TITLE
Pants Monorepo Support with sphinx-collections?

### DISCUSSION THREAD

**Discussion by jgarte** [useblocks/sphinx-collections#8] [Ideas]
What do you think of integrating sphinx-collections with pants?

See this discussion:

https://github.com/pantsbuild/pants/discussions/17822

**Comment by danwos** [comment]
I haven't used Pants, so I have currently no idea what an integration could look like.

Shall Sphinx-Collections provides some config-files, so that it can be used and maybe configured inside Pants as well? But it still can be used standalone?
Or shall Sphinx-Collections be based on Pants, so that Pants becomes a dependency and must be installed and configured every time?

**Reply by jgarte** [reply]
> Or shall Sphinx-Collections be based on Pants, so that Pants becomes a dependency and must be installed and configured every time?

Hi, probably not the latter as the usual design pattern in pants is for pants to wrap Python developer ecosystem tools but not for Python developer ecosystem tools to wrap the pants build system. In other words, I'm imagining that sphinx-collections would become integrated into pants and not the other way around.

> I haven't used Pants, so I have currently no idea what an integration could look like.

From the research I've done so far, I'm not sure yet how pants should handle documentation for a monorepo. 
It is an open question. That said, pants is designed for monorepos but it doesn't currently support building documentation for monorepos.
